### PR TITLE
Fix #160: Translating typedef of function pointer doesn't take line l…

### LINKED ISSUE #160: Translating typedef of function pointer doesn't take line length into consideration
The following code:

```c
typedef void (*CXInclusionVisitor)(CXFile included_file,
                                   CXSourceLocation* inclusion_stack,
                                   unsigned include_len,
                                   CXClientData client_data);
```

Is translated to:

```d
alias CXInclusionVisitor = void function(CXFile included_file, CXSourceLocation* inclusion_stack, uint include_len, CXClientData client_data);
```

DStep should be able to split up the function pointer across multiple lines just as a regular function declaration.

### PR DESCRIPTION
**Pull request by ciechowoj**
…ength into consideration.

### CODE REVIEW

**Comment by ciechowoj** [comment]
Not that easy to fix as I thought initially...

**Comment by ciechowoj** [comment]
Updated. The problem with initial fix was it didn't handle nested function pointers (e.g. as a field of struct, see libclang unittest).

**Comment by jacob-carlborg** [comment]
Could you please give a brief description of the changes?

**Comment by ciechowoj** [comment]
> Could you please give a brief description of the changes?

It fixed the issue #160. The difficulty was with nested function pointers as their arguments and they themselves have to be intended. 
I have trouble with implementing it with the usage of `Output` class and `adaptiveLine`, I needed to add some proxy type `SourceNode` which could hold not only translated code but the information where the could could be broken.

**Comment by jacob-carlborg** [comment]
Ok, thanks.

File: unit_tests/FormattingUnitTest.d [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+void very_long_function_declaration(double way_too_long_argument,
+                           double another_long_argument);
+C",
+q"D
+extern (C):
+
+void very_long_function_declaration (double way_too_long_argument, double another_long_argument);
+D", options);
+
+}
```
[the remark this answers is not in the document]
Comment (ciechowoj) [reply unit_tests/FormattingUnitTest.d]: They weren't, but they are now. Updated the test as well.

[the remark this answers is not in the document]
Comment (ciechowoj) [reply unit_tests/FormattingUnitTest.d]: At least they will be after I push the changes... sigh...

**Comment by jacob-carlborg** [comment]
👌

**Comment by jacob-carlborg** [comment]
Good to go?

**Comment by ciechowoj** [comment]
Yeah.

**Comment by jacob-carlborg** [comment]
Thanks.

**Pull request merged**